packages/flutter/lib/src/material/dialog.dart

```diff
@@ -457,13 +457,18 @@ class _DialogRoute<T> extends PopupRoute<T> {
 
 /// Displays a dialog above the current contents of the app.
 ///
-/// This function typically receives a [Dialog] widget as its child argument.
-/// Content below the dialog is dimmed with a [ModalBarrier].
+/// This function typically receives a [Dialog] widget inside of a [WidgetBuilder]
+/// as it's `builder` argument. Content below the dialog is dimmed with a [ModalBarrier].
+/// This widget does not share a context with the location that `showDialog` is
+/// originally called from, nor will calling `setState()` in this location cause
+/// the dialog to update.
```

Comment:
Maybe say "Use a [StatefulBuilder] or a custom [StatefulWidget] if the dialog needs to update dynamically."

Reply:
Done

Reply:
Done